### PR TITLE
Update Apache

### PR DESCRIPTION
**Pull request by stevenelleman-okta**
Associated Issue: https://oktainc.atlassian.net/browse/OKTA-198383
1. Update httpd to 2.4.35, apr to 1.6.5, and apr-util 1.6.1 
2. Change hashes from md5 to sha256 
3. Make necessary dev_package additions

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,18 +4,18 @@ FROM gliderlabs/alpine:latest
...
+ENV APRUTIL_HASH d3e12f7b6ad12687572a3a39475545a072608f4ba03a6ce8a3778f607dd0035b  apr-util-1.6.1.tar.bz2
 
-
-ENV DEV_PACKAGES autoconf clang make wget build-base file musl-dev openssl-dev pcre-dev curl-dev sqlite-dev luajit-dev perl
-ENV RUNTIME_PACKAGES pcre openssl curl sqlite luajit
+ENV DEV_PACKAGES autoconf clang make wget build-base file musl-dev libressl-dev pcre-dev curl-dev sqlite-dev luajit-dev perl expat-dev
+ENV RUNTIME_PACKAGES pcre libressl curl sqlite luajit
 
 RUN mkdir /build
+RUN echo -e "http://nl.alpinelinux.org/alpine/v3.5/main\nhttp://nl.alpinelinux.org/alpine/v3.5/community" > /etc/apk/repositories
```
Comment (paulquerna-okta) [inline Dockerfile:18]: is there a reason this is needed?

Comment (stevenelleman-okta) [reply Dockerfile:18]: I would need to double check but I recall it fixing an issue regarding some of the other dev_packages.

Comment (stevenelleman-okta) [reply Dockerfile:18]:
```
fetch http://alpine.gliderlabs.com/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://alpine.gliderlabs.com/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  /bin/sh (virtual):
    provided by: busybox
    required by:
                 ca-certificates-20171114-r3[/bin/sh]
                 alpine-baselayout-3.1.0-r0[/bin/sh]
                 ca-certificates-20171114-r3[/bin/sh]
                 alpine-baselayout-3.1.0-r0[/bin/sh]
                 ca-certificates-20171114-r3[/bin/sh]
The command '/bin/sh -c apk --update add ${DEV_PACKAGES} ${RUNTIME_PACKAGES}' returned a non-zero code: 6
``` 

Code 6 = Host public key is unknown.

Comment (stevenelleman-okta) [reply Dockerfile:18]: My original addition was definitely a hack, so I'll work on a better solution. It looks like I can either disable host key checking (which would leave us vulnerable) or add the host public key for `alpine.gliderlabs.com`

Comment (stevenelleman-okta) [reply Dockerfile:18]: Well it looks like http://alpine.gliderlabs.com/alpine/  isn't secure in the first place, so I'll disable host key checking.

Comment (stevenelleman-okta) [reply Dockerfile:18]: Hm, so changing `RUN apk --update add ${DEV_PACKAGES} ${RUNTIME_PACKAGES}` to `RUN apk update add ${DEV_PACKAGES} ${RUNTIME_PACKAGES}` corrected the issue. Not sure why.

**Comment by stevenelleman-okta** [comment]
The new problem: 
```
checking for chosen layout... apr
checking for gcc... /usr/bin/clang
checking whether the C compiler works... no
configure: error: in `/build/apr-1.6.5':
configure: error: C compiler cannot create executables
See `config.log' for more details
The command '/bin/sh -c ./configure --prefix=/opt --enable-nonportable-atomics --enable-threads' returned a non-zero code: 77
```

**Pull request closed without merging**